lib/vdi.py

```diff
@@ -89,3 +93,9 @@ def param_clear(self, param_name):
     def param_remove(self, param_name, key, accept_unknown_key=False):
         _param_remove(self.sr.pool.master, self.xe_prefix, self.uuid,
                       param_name, key, accept_unknown_key)
+
+    def wait_for_coalesce(self):
+        # It is necessary to wait a long time because the GC can be paused for more than 5 minutes.
+        # And it is also necessary to allow a sufficiently long merge time which depends on the amount of data.
+        wait_for_not(lambda: self.get_parent(), msg="Waiting for coalesce", timeout_secs=7 * 60)
```

Comment:
It should be noted that it will only work if we are on depth=2 (i.e. only one snapshot and we are the leaf).
If we have multiple snapshot, the parent will be the grandparent instead, resulting in a timeout here but the coalesce did happen.